source/camera/SlopeWalk.cpp

```diff
@@ -0,0 +1,321 @@
+/*
+ *  Copyright (C) 2011-2015  OpenDungeons Team
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "camera/SlopeWalk.h"
+#include "utils/LogManager.h"
+
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/point_xy.hpp>
+#include <boost/geometry/geometries/polygon.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/geometry/geometries/adapted/boost_tuple.hpp>
+
+using namespace boost::geometry;
+
+BOOST_GEOMETRY_REGISTER_POINT_2D(Vector3i, int64_t, boost::geometry::cs::cartesian, x, y)
+//! Values used to know whether to show and/or hide a mesh
+typedef Vector3i Point;
+typedef model::polygon<Point> Polygon;
+
+template <typename Point>
+void get_coordinates(Point const& p)
+{
+    using boost::geometry::get;
+    std::stringstream ss;
+    ss << "get__coordinates : " ; 
+    ss << double(get<0>(p))/Unit << ", " << double(get<1>(p))/Unit << std::endl;
+    LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL,__FILE__,__LINE__,ss.str());
+}
+
+
+
+
+SlopeWalk::SlopeWalk():
+    mLeftVertexPassed(false),
+    mRightVertexPassed(false)
+{
+}
+
+void SlopeWalk::buildSlopes()
+{
+    int ii;
```

Comment:
Yes but the stop condition is ii != mDownLeftIndex . That means that pushing mDownLeftIndex is enough.